spec/folio_sync/archives_space/marc_exporter_spec.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe FolioSync::ArchivesSpace::MarcExporter do
+  let(:instance) { described_class.new }
+  let(:client) { instance_double(FolioSync::ArchivesSpace::Client) }
+  let(:logger) { instance_double(Logger, info: nil) }
+
+  before do
+    allow(FolioSync::ArchivesSpace::Client).to receive(:instance).and_return(client)
+    allow(Logger).to receive(:new).and_return(logger)
+  end
+
+  describe '#initialize' do
+    it 'can be instantiated' do
+      expect(instance).to be_a(described_class)
+    end
+
+    it 'initializes with the ArchivesSpace client' do
+      synchronizer = described_class.new
+      expect(synchronizer.instance_variable_get(:@client)).to eq(client)
+    end
+
+    it 'initializes with a logger' do
+      expect(instance.instance_variable_get(:@logger)).to eq(logger)
+    end
+  end
+
+  describe '#export_recent_resources' do
+    let(:repositories) do
+      [
+        { 'uri' => '/repositories/1', 'publish' => true },
+        { 'uri' => '/repositories/2', 'publish' => false }
+      ]
+    end
+    let(:modified_since) { Time.utc(2023, 1, 1) }
+
+    before do
+      allow(client).to receive(:get_all_repositories).and_return(repositories)
+      allow(instance).to receive(:log_repository_skip)
+      allow(instance).to receive(:export_resources_from_repository)
+    end
+
+    it 'fetches all repositories from the ArchivesSpace client' do
+      instance.export_recent_resources(modified_since)
+      expect(client).to have_received(:get_all_repositories)
+    end
+
+    it 'processes published repositories' do
+      instance.export_recent_resources(modified_since)
+      expect(instance).to have_received(:export_resources_from_repository).with('1', modified_since)
+    end
+
+    it 'skips unpublished repositories' do
+      instance.export_recent_resources(modified_since)
+      expect(instance).to have_received(:log_repository_skip).with(repositories[1])
+    end
+  end
+
+  describe '#export_resources_from_repository' do
+    let(:repo_id) { '1' }
+    let(:resources) { [{ 'uri' => '/resources/1', 'title' => 'Resource 1', 'id' => '1', 'identifier' => '123' }] }
+    let(:query_params) { { q: 'test_query', page: 1, page_size: 20 } }
+    let(:modified_since) { Time.utc(2023, 1, 1) }
+
+    before do
+      allow(instance).to receive(:build_query_params).and_return(query_params)
+      allow(client).to receive(:retrieve_paginated_resources).and_yield(resources)
+      allow(instance).to receive(:log_resource_processing)
+      allow(instance).to receive(:export_marc_for_resource)
+    end
+
+    it 'builds query parameters with the provided modified_since timestamp' do
+      allow(Time).to receive(:now).and_return(Time.utc(2025, 4, 1, 12, 0o0, 0o0, 123_939))
```

Comment:
I'm actually not familiar with `0o0` number notation in Ruby!  Why use `0o0` rather than `0`?

Reply:
I had entered the time as "00" which got converted to "0o0" (octal) by Rubocop. I just fixed it to use 0. Thanks for catching that!